bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -16,43 +16,59 @@ export class FileInputView extends WidgetView {
     if (this.dialogEl == null) {
       this.dialogEl = document.createElement('input')
       this.dialogEl.type = "file"
-      this.dialogEl.multiple = false
-      this.dialogEl.onchange = (e) => this.load_file(e)
+      this.dialogEl.multiple = this.model.multiple
+      this.dialogEl.onchange = async (e) => await this.load_files(e)
       this.el.appendChild(this.dialogEl)
     }
     if (this.model.accept != null && this.model.accept != '')
       this.dialogEl.accept = this.model.accept
+
     this.dialogEl.style.width = `{this.model.width}px`
     this.dialogEl.disabled = this.model.disabled
   }
 
-  load_file(e: any): void {
-    const reader = new FileReader()
-    this.model.filename = e.target.files[0].name
-    reader.onload = (e) => this.file(e)
-    reader.readAsDataURL(e.target.files[0])
+  async load_files(e: any): Promise<void> {
+    const files = e.target.files
```

Comment:
A lot of this code was originally written without proper care, but with the extent of changes in this PR, let's do this properly this time and actually fill in those gaps. When `e` is of type `any` then all its properties are of type `any` and so on, so we are back with dynamically typed JavaScript, and that's the last we want. `e` comes from `onchange` callback, thus we can only use the not so specific `Event` type. Though not super useful, it will already show a flaw with this code, mainly `e.target` can be `null`. You could deal with this, but it would be even better to pass `files` (of type `FileList`) directly to this method, especially given that you don't use `e` for any other purpose:
```ts
async load_files(files: FileList): Promise<void> { ... }
```
Given that we really don't rely on the event target mechanism altogether, we can use `this.dialogEl.files` directly:
```ts
this.dialogEl.onchange = async () => {
  const {files} = this.dialogEl
  if (files != null) {
    await this.load_files(files)
  }
}
```
`files` can be `null`, so we have to deal with this. I left `async` and `await`, but those aren't strictly necessary in this context, though it doesn't hurt to leave them either.

Reply:
Please, do not impute "carelessness" if the worker (me) notified you about a huge lack of competence and experience in TS at the very start of the comittements.
Anyway, thank you for the patience and proposals, I tried to follow with my latest commit.

Reply:
@bruns01 @mattpap is referring to the original state of the code from before your PR, and there is indeed lots of old technical debt we can clean up (which is why PRs like his are a very nice opportunity!)

Reply:
@bryevdv  Thanks, for that clarification!